model/otlpgrpc/logs.go

```diff
@@ -139,7 +140,36 @@ func (lr LogsRequest) MarshalJSON() ([]byte, error) {
 
 // UnmarshalJSON unmarshalls LogsRequest from JSON bytes.
 func (lr LogsRequest) UnmarshalJSON(data []byte) error {
-	return jsonUnmarshaler.Unmarshal(bytes.NewReader(data), lr.orig)
+	if err := jsonUnmarshaler.Unmarshal(bytes.NewReader(data), lr.orig); err != nil {
+		return err
+	}
+	lr.instrumentationLibraryLogsToScope()
+	return nil
+}
+
+// instrumentationLibraryLogsToScope implements the translation of resource logs data
+// following the v0.15.0 upgrade:
+//    	receivers SHOULD check if instrumentation_library_logs is set
+// 		and scope_logs is not set then the value in instrumentation_library_logs
+// 		SHOULD be used instead by converting InstrumentationLibraryLogs into ScopeLogs.
+// 		If scope_logs is set then instrumentation_library_logs SHOULD be ignored.
+func (lr LogsRequest) instrumentationLibraryLogsToScope() {
```

Comment:
Make this a func that accepts a slice of `ResourceLogs` and call it from the otlp.JsonUnmarshaler as well.

Reply:
Done 👍